crates/sparrow-syntax/src/syntax/signature.rs

```diff
@@ -39,9 +88,23 @@ impl Signature {
             )
         }
 
+        // check that no duplicates exist in the type_parameters
+        let mut names = HashSet::new();
+        for type_param in &type_parameters {
+            if !names.insert(&type_param.name.0) {
+                anyhow::bail!(
+                    "Duplicate type parameter '{}' in signature for '{}'",
+                    type_param.name.0,
+                    name
+                )
+            }
+        }
+        std::mem::drop(names);
```

Comment:
I tend to avoid `std::mem::drop` for this "eagerly drop it". Why not put it in a block? Also, it will be dropped as soon as this function returns, which is the next thing that happens, so ti doesn't seem to add much.